### PR TITLE
Integrate places with events

### PR DESCRIPTION
**Pull request by zhoua0805**
- Allow users to add an event from places
- Render an input form for event duration at cursor location
- change the add button icon (and enable/disable it) when a place is added as an event/ an event is deleted

Quick Demo: 
image

### CODE REVIEW

File: planet/src/main/webapp/css/itinerary.css [hunk 1]
```diff
@@ -47,6 +47,14 @@ h1,h2,h3,h4,h5,h6{
     z-index: 999;
 }
 
+#add-place-to-event-form {
+    position: absolute;
+    display: none;
+    background: white;
+    border: solid 1px;
```
Comment (emilyzhaoo) [inline planet/src/main/webapp/css/itinerary.css:54]: Instead of px to make the website respondent, try using vh or vw

**Review by emilyzhaoo — APPROVED** [review]
LGTM with nit

File: planet/src/main/webapp/js/placesItinerary.js [hunk 2]
```diff
@@ -46,3 +59,71 @@ function createPlaceElement(ref, name, address) {
...
+        openingTime: placeDetails.openingTime,
+        closingTime: placeDetails.closingTime,
+        order: order,
+    });
+    closeAddPlaceForm();
+    renderPlaces();
+}
+
+function validatePlaceDuration(duration) {
+    let isValid = true;
```
Comment (EkkiD) [inline planet/src/main/webapp/js/placesItinerary.js]: it looks like this variable isn't needed. Anytime it's set to false, we immediately return. We can just directly return false or true instead.

**Review by EkkiD — CHANGES_REQUESTED** [review]

**Review by azhang72 — APPROVED** [review]
lgtm pending EkkiD's comments

File: planet/src/main/webapp/css/itinerary.css (hunk 1 above)
Comment (zhoua0805) [reply planet/src/main/webapp/css/itinerary.css:54]: Okay!

File: planet/src/main/webapp/js/placesItinerary.js (hunk 2 above)
Comment (zhoua0805) [reply planet/src/main/webapp/js/placesItinerary.js]: Fixed!

**Review by EkkiD — APPROVED** [review]
LGTM

**Pull request merged**